### DISCUSSION TITLE
How isolate a agent to run a specific workflow?

### DISCUSSION THREAD

**Discussion by eliasscosta** [woodpecker-ci/woodpecker#3791] [Ask a Question]
Does anyone know if is possible to isolate an agent to run a specific workflow?  Without adding labels to the configuration in the workflow file. 

My case:  

My general workflows run in our k8s with the `k8s` backend. It's ok and runs as expected. But if I add a Mac and install the agent with `local` backend. The Mac receives workflows that need to run in the k8s backend. I try to add labels to the agent. But it only works if I put labels into all workflows across the repositories.

**Comment by zc-devs** [comment]
Might be related: #3483, #3539

**Comment by qwerty287** [comment]
I tihnk https://github.com/woodpecker-ci/woodpecker/pull/3483 will fix this
